pkg/ccrm/ccrm_handler.go

```diff
@@ -55,29 +76,112 @@ type NodeAttributesData struct {
 
-func (s *CCRMHandler) Init(ctx context.Context, nodeType string, nodeMgr *node.NodeMgr, caches *CCRMCaches, redisClient *redis.Client, ctrlConn *grpc.ClientConn, flags *Flags, registryAuth *cloudcommon.RegistryAuth) {
+func (s *CCRMHandler) Init(ctx context.Context, nodeMgr *node.NodeMgr, caches *CCRMCaches, platformBuilders map[string]platform.PlatformBuilder, flags *Flags, registryAuthAPI cloudcommon.RegistryAuthApi) {
 	s.caches = caches
 	s.nodeMgr = nodeMgr
-	s.redisClient = redisClient
-	s.ctrlConn = ctrlConn
+	s.nodeType = nodeMgr.MyNode.Key.Type
 	s.flags = flags
-	s.registryAuth = registryAuth
+	s.registryAuthAPI = registryAuthAPI
 	s.nodeAttributesCache.Init()
+	s.crmPlatforms.Init()
+	s.platformBuilders = platformBuilders
+	s.vaultClient = accessapi.NewVaultClient(ctx, nodeMgr.VaultConfig, s, flags.Region, flags.DnsZone, nodeMgr.ValidDomains)
+	s.cloudletSSHKey = cloudletssh.NewSSHKey(s.vaultClient)
+	s.crmHandler = crmutil.NewCRMHandler(s.getCRMCloudletPlatform, s.nodeMgr)
+	s.proxyCertsCache = certscache.NewProxyCertsCache(s.vaultClient)
 
-	// notify handlers
-	s.caches.CloudletCache.AddUpdatedCb(s.cloudletChanged)
 	s.caches.CloudletNodeCache.AddUpdatedCb(s.cloudletNodeChanged)
+	s.crmHandler.SettingsCache.AddUpdatedCb(s.crmHandler.SettingsChanged)
+}
 
-	if redisClient != nil {
-		// redis handlers
-		hctx, cancel := context.WithCancel(ctx)
-		s.CancelHandlers = cancel
-		server := rediscache.GetCCRMAPIServer(redisClient, nodeType, s)
-		server.Start(hctx)
+func (s *CCRMHandler) InitConnectivity(client *notify.Client, kvstore objstore.KVStore, nodeMgr *node.NodeMgr, grpcServer *grpc.Server, sync edgeproto.DataSync) {
+	// Caches are updated via etcd watches.
+	// In general most data sent from here to the Controller is
+	// via return values from GRPC API calls the Controller makes to here.
+	s.sync = sync
+	s.crmHandler.AppCache.InitSync(sync)
+	s.crmHandler.AppInstCache.InitSync(sync)
+	s.crmHandler.CloudletInternalCache.InitSync(sync)
+	s.crmHandler.CloudletCache.InitSync(sync)
+	s.crmHandler.VMPoolCache.InitSync(sync)
+	s.crmHandler.FlavorCache.InitSync(sync)
+	s.crmHandler.ClusterInstCache.InitSync(sync)
+	s.crmHandler.TrustPolicyCache.InitSync(sync)
+	s.crmHandler.TrustPolicyExceptionCache.InitSync(sync)
+	s.crmHandler.AutoProvPolicyCache.InitSync(sync)
+	s.crmHandler.AutoScalePolicyCache.InitSync(sync)
+	s.crmHandler.SettingsCache.InitSync(sync)
+	s.crmHandler.ResTagTableCache.InitSync(sync)
+	s.crmHandler.GPUDriverCache.InitSync(sync)
+	s.crmHandler.AlertPolicyCache.InitSync(sync)
+	s.crmHandler.NetworkCache.InitSync(sync)
+	s.caches.PlatformFeaturesCache.InitSync(sync)
+	s.caches.CloudletNodeCache.InitSync(sync)
+	s.caches.CloudletInfoCache.InitSync(sync)
+	s.caches.ClusterInstInfoCache.InitSync(sync)
+	s.caches.AppInstInfoCache.InitSync(sync)
+	nodeMgr.CloudletLookup.GetCloudletCache(node.NoRegion).InitSync(sync)
+	nodeMgr.CloudletPoolLookup.GetCloudletPoolCache(node.NoRegion).InitSync(sync)
```

Comment:
Nit - I know that region is not used here, since we are using region cache, but just to make it clear, it's probably a good idea to pass the region(`nodeMgr.Region`).